addons/hr_expense/models/hr_expense.py

```diff
@@ -857,6 +856,19 @@ def _prepare_move_lines_vals(self):
                 account = self.product_id.product_tmpl_id._get_product_accounts()['expense']
             else:
                 account = self.env['ir.property']._get('property_account_expense_categ_id', 'product.category')
+
+            if not account:
```

Comment:
this condition is not needed ?

Reply:
yep needed for the 4th default mentioned in the PR message 
I have added clarifying comments as well